### PR TITLE
Mac build

### PR DESCRIPTION
**Pull request by eukarpov**

### CODE REVIEW

**Comment by shock** [comment]
@eukarpov **Update**: I have forked your repo (https://github.com/shock/openvino-plugins-ai-audacity) and cloned your mac-build branch to my own branch.  I am about to go through your work. I was experimenting with workflow in case I make any changes and I accidentally created a PR on your repo.  I closed it.  Now I think I understand what happened to you yesterday.  :)

**Comment by shock** [comment]
@eukarpov 

How are you running the build locally?  Or are you running it only on GitHub?  I tried using `act` to test locally:

```
act -s GITHUB_TOKEN=$ACT_GITHUB_TOKEN -P macos-14=-self-hosted -j build-openvino-plugins
```

But it doesn't check out the this repository into the workspace properly.  

So, I resorted to building manually following the steps in your [`build.sh`](https://github.com/eukarpov/openvino-plugins-ai-audacity/pull/1/files#diff-24391a00a3942f0797db5f0dc44e57d3ec1585f213ed8a95ce2d8c8ee710fa1d) without using XCode IDE.  I had to tweak a few things to get CMake to build an `arm64` target for Audacity, but it succeeded.  When I run Audacity, I see mod-openvino in the Preferences modules list and I set it to "Enabled".  But similar to @cshaver, when I restart Audacity, I get 

image

I followed all of your steps in https://github.com/intel/openvino-plugins-ai-audacity/issues/28#issuecomment-2022884464.  When I run the `xattr` command on the .so and .dylib from `m_openvino_toolkit_macos_11_0_2024.0.0.14509.34caeefd078_arm64`, I get `No such xattr: com.apple.quarantine`, so I don't think that's the problem.  

I copied all of the required OpenVINO models from huggingface into `build/audacity/Release/Audacity.app/Contents/openvino-models`.

I even tried using [your artifacts](https://github.com/eukarpov/openvino-plugins-ai-audacity/actions/runs/8561925816?pr=1) , by overwriting mine with yours, but the problem persists.

In short, I am unable to help verify your build steps.  Maybe I am missing something.

Specs:
* 2021 M1 Mini
* MacOS 14.4.1
* CMake 3.29.1

Thanks.

**Comment by eukarpov** [comment]
It is exactly what should be resolved now. It was possible to get the local build and configure it locally. However, with GitHUB build, it is not working yet.  Expecting to fix it sometime this week.

**Comment by eukarpov** [comment]
It seems complicated to build only the module and deploy it to an existing Audacity app on macOS.
"Modules should be built on the same day as Audacity"
https://manual.audacityteam.org/man/modules_preferences.html

For now, it will be a full build with Audacity app.

**Comment by shock** [comment]
> It seems complicated to build only the module and deploy it to an existing Audacity app on macOS. "Modules should be built on the same day as Audacity" https://manual.audacityteam.org/man/modules_preferences.html
> 
> For now, it will be a full build with Audacity app.

Good find, and good to know.  My modules were all built together with Audacity, and mod-openvino.so was the only one not working.  The others work fine, so I assumed I was missing something else.  I will try again.

In the meantime, I tried your full build Audacity.app artifact from [your last GitHub push](https://github.com/eukarpov/openvino-plugins-ai-audacity/actions/runs/8624079438).  When I unzip it and run it, I get a vanilla error dialogue:

image

I investigated the .app folder's contents I don't see anything obviously wrong.  Did you try downloading it and running it from GitHub yourself?  

I'll work on trying to get it to build locally again when I get some more time.  Any advice you want to share about your local build environment would be helpful.  I'll share my environment when I get back to it.

**Comment by eukarpov** [comment]
This work is still in progress, and it is not working yet. However, the direction is to use the full bundle instead of attempting to deploy only the module.
